compiler/fir/java/src/org/jetbrains/kotlin/fir/java/enhancement/javaTypeUtils.kt

```diff
@@ -330,5 +330,7 @@ internal fun List<FirAnnotationCall>.computeTypeAttributesForJavaType(): ConeAtt
             CompilerConeAttributes.EnhancedNullability.ANNOTATION_CLASS_ID -> add(CompilerConeAttributes.EnhancedNullability)
             // TODO: Need to handle others too? E.g., COMPATQUAL_NONNULL_ANNOTATION_ID
             in NOT_NULL_ANNOTATION_IDS -> add(CompilerConeAttributes.EnhancedNullability)
+            JAVAX_NONNULL_ANNOTATION_ID -> add(CompilerConeAttributes.EnhancedNullability)
+            // TODO: What about nullable annotations?
```

Comment:
Should we add `@FlexibleNullability` for `@Nullable` annotations?

And, what about `javax...CheckForNull`? That sounds like we need `@EnhancedNullability` and add implicit cast to materialize null check?

Reply:
As I know, all Java not-null annotations should provoke `@EnhancedNullability`. About `@FlexibleNullability`, I think it should be only on flexible types when bounds nullability is different.

Reply:
Thanks to this comment, I completely rework how those attributes are added.

In the 1st commit, as a separate work, more Java non-null annotations are converted.

In the 2nd commit, instead of checking, e.g., if there isn't any other attributes, or if this is a platform type, I visited the places where _flexible_ type _with_ different nullability lower/upper bounds are generated, and added `@FlexibleNullability` to upper bound (where `nullable` is flipped).

While doing so, I relied on `union` of attributes, which seems like a natural way to accumulate attributes, and thanks to that, I found hidden bugs of that logic.